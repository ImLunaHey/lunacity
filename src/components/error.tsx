import React from 'react';
import { Text } from '@nextui-org/react';

const ErrorComponent: React.FC<{ statusCode: number; message: string }> = ({
  message,
}) => {
  return (
    <div
      className="relative rounded border border-red-500 bg-red-500 px-4 py-3 text-white"
      role="alert"
    >
      <span className="block sm:inline">{message}</span>
    </div>
  );
  //   return <Text color="error">{message}</Text>;
};

export default ErrorComponent;
