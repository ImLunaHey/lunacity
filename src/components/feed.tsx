import React from 'react';
import { SinglePost } from './single-post';
import type { SinglePostProps } from './single-post';
import InfiniteScroll from 'react-infinite-scroll-component';
import { Loading } from '@nextui-org/react';
import { useTranslation } from 'react-i18next';

type FeedProps = {
  items?: SinglePostProps[];
  publicCursor?: string;
  personalCursor?: string;
  className?: string;
  fetchData?: (opts?: { personalCursor?: string; publicCursor?: string }) => () => void;
};

const getUniqueItemsInArray = (items: SinglePostProps[]): Exclude<SinglePostProps, null>[] => {
  const seen: Record<string, boolean> = {};
  return (items as Exclude<SinglePostProps, null>[]).filter((item) =>
    seen.hasOwnProperty(item.id) ? false : (seen[item.id] = true)
  );
};

const Feed: React.FC<FeedProps> = ({ items, fetchData = (opts) => () => undefined, personalCursor, publicCursor }) => {
  const { t } = useTranslation();
  if (!items) return null;
  const uniqueItems = getUniqueItemsInArray(items);

  return (
    <InfiniteScroll
      next={fetchData({ publicCursor, personalCursor })}
      dataLength={uniqueItems.length}
      hasMore={uniqueItems.length === 50}
      loader={<Loading />}
      endMessage={
        <p style={{ textAlign: 'center' }}>
          <b>{t('no-more-posts')}</b>
        </p>
      }
      // // below props only if you need pull down functionality
      // refreshFunction={fetchData()}
      // pullDownToRefresh
      // pullDownToRefreshThreshold={50}
      // pullDownToRefreshContent={
      //   <h3 style={{ textAlign: 'center' }}>&#8595; Pull down to refresh</h3>
      // }
      // releaseToRefreshContent={
      //   <h3 style={{ textAlign: 'center' }}>&#8593; Release to refresh</h3>
      // }
    >
      {uniqueItems.map((post) => (
        <SinglePost post={post} key={post.id} />
      ))}
    </InfiniteScroll>
  );
};

export default Feed;
