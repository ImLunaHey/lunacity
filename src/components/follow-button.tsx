'use client';

import { Button, Loading } from '@nextui-org/react';
import { type FC, useState } from 'react';
import { api } from '../utils/api';
import ErrorComponent from '../components/error';
import { useSession } from 'next-auth/react';
import { type TRPCClientErrorLike } from '@trpc/client';
import { useTranslation } from 'react-i18next';

type handleErrorProps = {
  error: TRPCClientErrorLike<any>;
  setError: (error: { statusCode: number; message: string } | null) => void;
};

const handleError = ({ error, setError }: handleErrorProps) => {
  setError({ statusCode: 500, message: error.message });
  setTimeout(() => {
    setError(null);
  }, 10_000);
};

export const FollowButton: FC<{ handle: string }> = ({ handle }) => {
  const { t } = useTranslation();
  const session = useSession();
  const followingState = api.page.followingState.useQuery({ handle }, { enabled: session.status === 'authenticated' });
  const followPage = api.page.followPage.useMutation();
  const unfollowPage = api.page.unfollowPage.useMutation();
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<{
    statusCode: number;
    message: string;
  } | null>(null);

  const onError = (error: TRPCClientErrorLike<any>) => {
    handleError({ error, setError });
    setIsLoading(false);
  };
  const onSuccess = () => {
    // @TODO: handle if this fails fetching
    void followingState.refetch().then(() => {
      setIsLoading(false);
    });
  };

  // @TODO: This needs a debounce
  const onPress = () => {
    setIsLoading(true);

    // Unfollow page
    if (followingState.data) {
      unfollowPage.mutate(
        { handle },
        {
          onError,
          onSuccess,
        }
      );
    } else {
      // Follow page
      followPage.mutate(
        { handle },
        {
          onError,
          onSuccess,
        }
      );
    }
  };

  const generateText = (isFollowing?: boolean) => {
    if (isFollowing === undefined) return;
    return isFollowing ? t('unfollow') : t('follow');
  };

  // If we hit an error show it
  if (error) return <ErrorComponent {...error} />;

  if (session.status === 'loading') return <Loading />;

  // Unauthenticated users cannot follow/unfollow
  if (session.status === 'unauthenticated') return null;

  // Show the follow button
  return (
    <Button
      className="min-w-fit"
      size="xs"
      color="secondary"
      disabled={isLoading}
      aria-label={generateText(followingState.data)}
      onPress={onPress}
    >
      {generateText(followingState.data)}
    </Button>
  );
};
