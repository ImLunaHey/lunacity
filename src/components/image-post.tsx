import type { SinglePostProps } from '@app/components/single-post';
import { TagCloud } from '@app/components/tag-cloud';
import { PageAvatar } from '@app/components/page-avatar';
import { Card, Grid, Button, Text } from '@nextui-org/react';
import { useTranslation } from 'react-i18next';

export const ImagePost: React.FC<{ post: SinglePostProps }> = ({ post }) => {
  const { t } = useTranslation();
  if (!post) return null;

  return (
    <Card css={{ p: '$6' }}>
      <Card.Header>
        <PageAvatar
          // @TODO: Fix this
          official={false}
          src={'https://nextui.org/images/card-example-5.jpeg'}
          name={post.page.displayName}
          handle={post.page.handle}
          description={post.page.description}
          followerCount={post.page.followerCount}
          followingCount={post.page.followingCount}
          popover={true}
        />
        <Grid.Container css={{ pl: '$6' }}>
          <Grid xs={12}>
            <Text h4>{post.page.displayName}</Text>
          </Grid>
        </Grid.Container>
      </Card.Header>
      <Card.Body css={{ py: '$2' }}>
        <div style={{ position: 'relative', width: '100%' }}>{/* <Image {...image} alt={image.alt} /> */}</div>
        {/* {body && (<ReactMarkdown remarkPlugins={[remarkGfm]}>{body}</ReactMarkdown>)} */}

        <Grid.Container alignItems="flex-start" gap={0.5}>
          <Grid>
            <TagCloud tags={post.tags} />
          </Grid>
        </Grid.Container>
      </Card.Body>
      <Card.Footer>
        <Grid.Container alignItems="flex-start" gap={0.5}>
          <Grid>
            <Button className="min-w-0" size="xs" color="default" aria-label={t('bookmark-post')}>
              üîñ
            </Button>
          </Grid>
          <Grid>
            <Button className="min-w-0" size="xs" color="default" aria-label={t('add-reaction')}>
              ‚≠ê
            </Button>
          </Grid>
          <Grid>
            <Button className="min-w-0" size="xs" aria-label={t('post-add-comment')}>
              üí¨
            </Button>
          </Grid>
        </Grid.Container>
      </Card.Footer>
    </Card>
  );
};
