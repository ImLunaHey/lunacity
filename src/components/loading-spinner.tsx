import type { FC } from 'react';
import { Loading } from '@nextui-org/react';

export const LoadingSpinner: FC = () => {
  return <Loading type="points-opacity" color="currentColor" size="sm" />;
};
