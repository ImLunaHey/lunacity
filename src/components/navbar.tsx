import { useSSR } from '@nextui-org/react';
import React, { type FC } from 'react';
import { Navbar, Text, Input, Dropdown, Avatar } from '@nextui-org/react';
import { signOut, useSession } from 'next-auth/react';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { SearchIcon } from '../icons/search-icon';
import { Notifications } from './notifications';
import { useTranslation } from 'react-i18next';
import { type SubmitHandler, useForm } from 'react-hook-form';
import Logo from '@app/components/logo';

const DropdownMenu: FC = () => {
  const { data: session } = useSession();
  const { t } = useTranslation(['common']);
  if (!session?.user?.page?.handle) {
    return (
      <Dropdown.Menu aria-label="User menu actions" color="secondary">
        <Dropdown.Item textValue="Create page" key="profile" className="p-0">
          <Link className="m-0 block h-full w-full p-2 text-white" href="/page/create">
            <Text b color="inherit" css={{ d: 'flex' }}>
              Create page
            </Text>
          </Link>
        </Dropdown.Item>
        <Dropdown.Item command="⌘S" textValue="Settings" key="settings" withDivider>
          <Link className="m-0 block h-full w-full text-gray-400 hover:text-white" href="/settings">
            {t('page.settings.title')}
          </Link>
        </Dropdown.Item>
        <Dropdown.Item textValue="Help and feedback" key="help_and_feedback">
          <Link className="m-0 block h-full w-full text-gray-400 hover:text-white" href="/help">
            {t('page.help.title')}
          </Link>
        </Dropdown.Item>
        <Dropdown.Item textValue="Signout" key="signout" withDivider color="error">
          <a
            className="m-0 block h-full w-full text-gray-400 hover:text-white"
            onClick={() => void signOut({ callbackUrl: '/' })}
          >
            {t('signout')}
          </a>
        </Dropdown.Item>
      </Dropdown.Menu>
    );
  }

  return (
    <Dropdown.Menu aria-label="User menu actions" color="secondary">
      <Dropdown.Item textValue={`@${session.user?.page?.handle}`} key="profile" className="p-0">
        <Link className="m-0 block h-full w-full p-2 text-white" href={`/${session.user?.page?.handle}`}>
          <Text b color="inherit" css={{ d: 'flex' }}>
            @{session.user?.page?.handle}
          </Text>
        </Link>
      </Dropdown.Item>
      <Dropdown.Item command="⌘S" textValue="Settings" key="settings" withDivider>
        <Link className="m-0 block h-full w-full text-gray-400 hover:text-white" href="/settings">
          {t('page.settings.title')}
        </Link>
      </Dropdown.Item>
      <Dropdown.Item textValue="Analytics" key="analytics">
        <Link className="m-0 block h-full w-full text-gray-400 hover:text-white" href="/analytics">
          {t('page.analytics.title')}
        </Link>
      </Dropdown.Item>
      <Dropdown.Item textValue="Help and feedback" key="help_and_feedback">
        <Link className="m-0 block h-full w-full text-gray-400 hover:text-white" href="/help">
          {t('page.help.title')}
        </Link>
      </Dropdown.Item>
      <Dropdown.Item textValue="Signout" key="signout" withDivider color="error">
        <a
          className="m-0 block h-full w-full text-gray-400 hover:text-white"
          onClick={() => void signOut({ callbackUrl: '/' })}
        >
          {t('signout')}
        </a>
      </Dropdown.Item>
    </Dropdown.Menu>
  );
};

const UserMenu: FC = () => {
  return (
    <Dropdown placement="bottom-right">
      <Navbar.Item>
        <Dropdown.Trigger>
          <Avatar
            bordered
            as="button"
            color="primary"
            size="md"
            src="https://i.pravatar.cc/150?u=a042581f4e29026704d"
          />
        </Dropdown.Trigger>
      </Navbar.Item>
      <DropdownMenu />
    </Dropdown>
  );
};

export default function NavBar() {
  const { data: session } = useSession();
  const router = useRouter();
  const { t } = useTranslation(['common']);
  const { isBrowser } = useSSR();

  // Search box
  type Inputs = { query: string };
  const { register, handleSubmit } = useForm<Inputs>();
  const onSubmit: SubmitHandler<Inputs> = (data) => {
    void router.push({
      pathname: '/search',
      query: {
        query: data.query,
      },
    });
  };

  // Don't render the navbar when we're within SSR
  // See: https://github.com/nextui-org/nextui/issues/779
  if (!isBrowser) return null;

  // If there's no session at all don't render
  if (!session) return null;

  return (
    <Navbar isBordered variant="sticky">
      <Navbar.Brand css={{ mr: '$4' }}>
        <Logo />
        <Link href="/">
          <Text b color="inherit" css={{ mr: '$11' }} hideIn="xs">
            Lunacity
          </Text>
        </Link>
        <Navbar.Content hideIn="xs" variant="highlight">
          <Navbar.Link isActive={router.asPath === '/'} href="/" as={Link}>
            🏡 {t('page.home.title')}
          </Navbar.Link>
          <Navbar.Link isActive={router.asPath === '/explore'} href="/explore" as={Link}>
            🌏 {t('page.explore.title')}
          </Navbar.Link>
          <Navbar.Link isActive={router.asPath === '/messages'} href="/messages" as={Link}>
            📩 {t('page.messages.title')}
          </Navbar.Link>
        </Navbar.Content>
      </Navbar.Brand>

      {/* Search box */}
      <Navbar.Content
        className="w-full"
        css={{
          ':first-child': {
            width: '100%',
          },
        }}
      >
        <Navbar.Item className="mr-3 w-full">
          {/* eslint-disable-next-line @typescript-eslint/no-misused-promises */}
          <form onSubmit={handleSubmit(onSubmit)}>
            <Input
              aria-label="Search box"
              clearable
              contentLeft={<SearchIcon fill="var(--nextui-colors-accents6)" size={16} />}
              contentLeftStyling={false}
              className="w-full"
              css={{
                '& .nextui-input-content--left': {
                  height: '100%',
                  marginLeft: '$4',
                  dflex: 'center',
                },
              }}
              placeholder="Search..."
              {...register('query', { required: true, maxLength: 25 })}
            />
          </form>
        </Navbar.Item>
      </Navbar.Content>
      <Navbar.Content
        css={{
          '@xsMax': {
            jc: 'space-between',
          },
        }}
      >
        <Notifications />
        <UserMenu />
      </Navbar.Content>
    </Navbar>
  );
}
