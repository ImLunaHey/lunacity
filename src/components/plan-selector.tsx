import { Card, Grid, Row, Text } from '@nextui-org/react';
import type { FC } from 'react';

export const PlanSelector: FC<{ hidden: boolean }> = ({ hidden }) => {
  const list = [
    {
      title: 'Orange',
      img: '/images/fruit-1.jpeg',
      price: '$5.50',
    },
    {
      title: 'Tangerine',
      img: '/images/fruit-2.jpeg',
      price: '$3.00',
    },
    {
      title: 'Cherry',
      img: '/images/fruit-3.jpeg',
      price: '$10.00',
    },
    {
      title: 'Lemon',
      img: '/images/fruit-4.jpeg',
      price: '$5.30',
    },
    {
      title: 'Avocado',
      img: '/images/fruit-5.jpeg',
      price: '$15.70',
    },
    {
      title: 'Lemon 2',
      img: '/images/fruit-6.jpeg',
      price: '$8.00',
    },
    {
      title: 'Banana',
      img: '/images/fruit-7.jpeg',
      price: '$7.50',
    },
    {
      title: 'Watermelon',
      img: '/images/fruit-8.jpeg',
      price: '$12.20',
    },
  ];

  if (hidden) return null;

  return (
    <Grid.Container gap={2} justify="flex-start">
      {list.map((item, index) => (
        <Grid xs={6} sm={3} key={index}>
          <Card isPressable>
            <Card.Body css={{ p: 0 }}>
              <Card.Image
                src={'https://nextui.org' + item.img}
                objectFit="cover"
                width="100%"
                height={140}
                alt={item.title}
              />
            </Card.Body>
            <Card.Footer css={{ justifyItems: 'flex-start' }}>
              <Row wrap="wrap" justify="space-between" align="center">
                <Text b>{item.title}</Text>
                <Text
                  css={{
                    color: '$accents7',
                    fontWeight: '$semibold',
                    fontSize: '$sm',
                  }}
                >
                  {item.price}
                </Text>
              </Row>
            </Card.Footer>
          </Card>
        </Grid>
      ))}
    </Grid.Container>
  );
};
