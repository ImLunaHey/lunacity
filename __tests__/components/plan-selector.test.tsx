import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { PlanSelector } from '@app/components/plan-selector';

describe('PlanSelector', () => {
  it("doesn't render when hidden", () => {
    const { container } = render(<PlanSelector hidden={true} />);
    expect(container).toMatchSnapshot();
  });

  it('renders when not hidden', () => {
    const { container } = render(<PlanSelector hidden={false} />);
    expect(container).toMatchSnapshot();
  });
});
