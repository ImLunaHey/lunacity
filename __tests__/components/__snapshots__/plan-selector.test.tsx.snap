// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanSelector doesn't render when hidden 1`] = `<div />`;

exports[`PlanSelector renders when not hidden 1`] = `
<div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-ienfGvA-css nextui-grid-item nextui-grid-container"
  >
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Orange"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-1.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Orange
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $5.50
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Tangerine"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-2.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Tangerine
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $3.00
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Cherry"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-3.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Cherry
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $10.00
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Lemon"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-4.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Lemon
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $5.30
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Avocado"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-5.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Avocado
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $15.70
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Lemon 2"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-6.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Lemon 2
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $8.00
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Banana"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-7.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Banana
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $7.50
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-ifdaEcH-css nextui-grid-item xs sm"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-eZMbDJ-variant-shadow nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-iZxJaC-isPressable-true"
        role="button"
        tabindex="0"
      >
        <div
          class="nextui-c-eFfoBo nextui-c-eFfoBo-ifBQDyP-css"
        >
          <div
            class="nextui-c-iVzbCs nextui-c-iVzbCs-bDGmTT-ready-true nextui-c-iVzbCs-iglXcVt-css nextui-image-container nextui-image--loading"
            data-state="loading"
          >
            <div
              class="nextui-c-FIiRH nextui-c-FIiRH-ibDGmTT-css nextui-image-skeleton"
            />
            <img
              alt="Watermelon"
              class="nextui-c-kbhVdb nextui-c-kbhVdb-ieCSSFU-css nextui-image"
              data-state="loading"
              height="140"
              src="https://nextui.org/images/fruit-8.jpeg"
              width="100%"
            />
          </div>
        </div>
        <div
          class="nextui-c-cakKrd nextui-c-cakKrd-ibVwQvh-css"
        >
          <div
            class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-ijHOHeI-css"
          >
            <b
              class="nextui-c-PJLV nextui-c-PJLV-ijnlmIz-css"
            >
              Watermelon
            </b>
            <p
              class="nextui-c-PJLV nextui-c-PJLV-igzIIoY-css"
            >
              $12.20
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
